nixos/modules/services/web-apps/collabora-online.nix

```diff
@@ -0,0 +1,1638 @@
+        handling = lib.mkOption {
+          type = lib.types.enum [
+            "report"
+            "log"
+            "both"
+            "ingore"
+          ];
+          default = "log";
+          description = "How to handle fonts mising in a document: 'report', 'log', 'both', or 'ignore'";
+        };
+      };
+
+      indirection_endpoint = {
+        url = lib.mkOption {
+          type = lib.types.str;
+          default = "";
+          description = ''URL endpoint to server which servers routeToken in json format.'';
+        };
+
+        migration_timeout_secs = lib.mkOption {
+          type = lib.types.ints.unsigned;
+          default = 180;
+          description = ''The maximum number of seconds waiting for shutdown migration message from indirection server before unloading an document. Defaults to 180 second.'';
+        };
+      };
+
+      zotero = {
+        enable = lib.mkOption {
+          type = lib.types.bool;
+          default = true;
+          description = ''Enable Zotero plugin.'';
+        };
+      };
+
+      help_url = lib.mkOption {
+        type = lib.types.str;
+        default = "https://help.collaboraoffice.com/help.html";
+        description = ''The Help root URL, or empty for no help (hides the Help buttons).'';
+      };
+
+      overwrite_mode = {
+        enable = lib.mkOption {
+          type = lib.types.bool;
+          default = true;
+          description = ''Enable overwrite mode (user can use insert key).'';
+        };
+      };
+
+      wasm = {
+        enable = lib.mkOption {
+          type = lib.types.bool;
+          default = false;
+          description = ''Enable WASM support.'';
+        };
+
+        force = lib.mkOption {
+          type = lib.types.bool;
+          default = false;
+          description = ''When enabled, all requests are redirected to WASM.'';
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users.cool = {
+      isSystemUser = true;
+      group = "cool";
+    };
+    users.groups.cool = { };
+
+    security.wrappers = {
+      coolforkit = {
+        owner = "root";
+        group = "root";
+        source = "${cfg.package}/bin/coolforkit";
+        capabilities = "cap_fowner,cap_chown,cap_sys_chroot=ep";
+      };
+
+      coolmount = {
+        owner = "root";
+        group = "root";
+        source = "${cfg.package}/bin/coolmount";
+        capabilities = "cap_sys_admin=ep";
+      };
+    };
+
+    systemd.services.coolwsd-systemplate-setup = {
+      description = "Collabora Online WebSocket Daemon Setup";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = utils.escapeSystemdExecArgs [
+          "${cfg.package}/bin/coolwsd-systemplate-setup"
+          "/var/lib/cool/systemplate"
+          "${cfg.package.libreoffice}/lib/collaboraoffice"
+        ];
+        RemainAfterExit = true;
+        StateDirectory = "cool";
+        Type = "oneshot";
+        User = "cool";
```

Comment:
Could this be run using `DynamicUser = true`?

Reply:
`coolwsd-systemplate-setup.service` shares the same data dir as `coolwsd.service`. Yes, it is possible to use `DynamicUser = true`, but that would mean that systemd will move/chown the state dir back and forth.

I.e., after running `coolwsd-systemplate-setup.service`: `/var/lib/cool` is a symlink to `private/cool`, owned by `coolwsd-systemplate`.
Then after running `coolwsd.service`: `/var/lib/cool` is a directory owned by `cool`.

Could you please elaborate on why we might want `DynamicUser` here?

Reply:
If there's state sharing, that's good enough of a reason not to use dynamic user.